### DISCUSSION TITLE
x-data no match in html element

### DISCUSSION THREAD

**Discussion by neapolis79** [alpinejs/alpine#1656] [7. Bugs (archived)] [answered]
hi, 
i'm upgrading my application from alpine 2 to alpune3...i read the upgrade guide on the site but i'm having a strange behavior....

originallu in my page i had 

`<html x-data="initTheme()" ...>`

and in the script section 
`<script>
function initTheme() {
return { theme: 'dark'} 
}`

now following the documentation i changed in 
`<html x-data="initTheme" ...>`

whith 
`<script>
            document.addEventListener('alpine:init', () => {
                Alpine.data('initTheme', () => ({
                    theme: 'dark'
                }))
            })
        </script>`

but i receive this error in console

Uncaught TypeError: el.matches is not a function
    at app.js:1874
    at Array.some (<anonymous>)
    at closestRoot (app.js:1874)
    at outNestedComponents (app.js:1860)
    at Array.filter (<anonymous>)
    at Object.start (app.js:1861)
    at Module../resources/js/app.js (app.js:4929)
    at __webpack_require__ (app.js:22403)
    at app.js:22553
    at Function.__webpack_require__.O (app.js:22440)

can you help me ? i didn't understand what is wrong

**Comment by KevinBatdorf** [comment, marked as answer]
`x-data` can't be on the `html` tag. Is there a specific reason you need it there? Typically the "Alpine way" is to put it only where you need it so that you don't spend time walking the DOM tracking everything, but if you just need to get upgraded, it should work on the body tag

**Reply by neapolis79** [reply]
ok thanks, originally was there because i needed to do some stuff on the head tag, but in the meantime i changed many things so now it's not needed anymore....so i will move this data directly on the button that use this data...thanks
